language-design.md

```diff
@@ -28,28 +102,114 @@ you know a poorly designed language? What are the symptoms?
+language. While there are certain principles that can help you spot a
+well-designed language, time may be a factor in discovering a languages
+shortcomings.
+
+A well designed language should be approchable and "self-documenting." Just like
+APIs you should be able to understand the code without constantly looking at
+documentation once you know the pattern. This leads me to a second point:
+well-designed languages must have an overall pattern as the first paper
+mentioned [Steele, 1998]. Furthermore, as Steele explained, language extenstions
+should feel natural and fit within the rest of the language. This provides a way
+of future-proofing a language. It also serves to make the language more readable
+and writable since patterns can be better maintained.
+
+Bad languages are either hard to read or hard to write. For example, in Verou's
+article regarding the choice of naming a "grayness" function, they raised the
+issue of calling such a function `gray` with a percentage value since it is
+not obviuous which means white and which means black. A lack of intuitiveness
+in language makes it bad; intuitiveness coupled with expected behavior can make
+it better.
+
+Efficiency with features is key to a well-designed language. It's not helpful
+to have 47 ways to express a for-loop since it makes the language inconsistent
+and harder to read. In the case of AppleScript (see the quote in the
+question below), there exist keywords solely to make things read more naturally,
+but this serves to make it harder to write. In should.js, a JavaScript test
+library that employs method chaining, one can add pointless keywords or omit
+them; this is an interesting idea, but does not necessarily make the language
+better. Here the flexibility has a trade-off with consistency of expression.
+
+Anthony slightly disagrees with the statement on should.js. He really likes
+the idea of a language where you can use optional statements to make it
+more readable without the "syntactic aspartame" being necessary.
+
+Well-designed languages should have errors in mind: not every program is going
+to be perfect, so there needs to be an elegant way of handling errors. It should
+not be an afterthought. However, this verges on prgamming language design versus
+compiler design.
 
 ---
 
 **Question**
-
 How do the themes of _Growing a Language_ relate to the "sound lab" we did this week?
 
 **Response**
 
-
+In _Growing a Language_, there was an emphais on "patterns": creating a language
+and use pattern is essential to usability, learnability, and extensibility. This
+relates to the other theme of fluency. This was especially exemplified by the
+entire talk defining a language made up of single syllable words. Steele
+describes that a language should be a mode of discussion between a user and the
+bytes that language creates.
+
+We felt that these themes were the main themes of the sound lab as well. One
```

Comment:
I made similar points in my response, that a solid pattern design is essential both on the user end and on the developer end. There should be a sense of fluency for the user, who should be able to use functions intuitively and familiarly. Similarly, if there is a good pattern in place, the developer should be able to grow the language and add new features with relative ease.